pkg/cli/sync.go

```diff
@@ -0,0 +1,159 @@
+)
+
+var (
+	_                        cli.Command = (*SyncCommand)(nil)
+	allowedSourceSystem                  = []string{"GOOGLEGROUP"}
+	allowedDestinationSystem             = []string{"GITHUB"}
+)
+
+type SyncCommand struct {
+	cli.BaseCommand
+
+	source             string
+	destination        string
+	groupMappingConfig string
+	userMappingConfig  string
+	sourceToken        string
+	destinationToken   string
+}
+
+func (c *SyncCommand) Desc() string {
+	return `Sync membership information`
+}
+
+func (c *SyncCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  Sync membership information from source and target system.
+
+  Sync membership from google group to GitHub
+
+  tlctl sync run \
+	-src GoogleGroup \
+	-dest GitHub \
+	-group-mapping-config your-group-config.textproto \
+	-user-mapping-config your-user-confif.textproto
+`
+}
+
+func (c *SyncCommand) Flags() *cli.FlagSet {
+	set := c.NewFlagSet()
+
+	// Command options
+	f := set.NewSection("COMMAND OPTIONS")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source",
+		Target:  &c.source,
+		Aliases: []string{"src", "s"},
+		Example: "GoogleGroup",
+		Usage:   `The source system to read membership `,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "destination",
+		Target:  &c.destination,
+		Aliases: []string{"d", "dst"},
+		Example: "GitHub",
+		Usage:   `The target system for membership information`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "group-mapping-config",
+		Target:  &c.groupMappingConfig,
+		Aliases: []string{"g", "gc", "group-config", "g-config"},
+		Example: "group-mapping-config.textproto",
+		Usage: `The group mapping config that contains group id mapping ` +
+			`from source system to destination system`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "user-mapping-config",
+		Target:  &c.userMappingConfig,
+		Aliases: []string{"u", "uc", "user-config", "u-config"},
+		Example: "user-mapping-config.textproto",
+		Usage: `The group mapping config that contains group id mapping ` +
+			`from source system to destination system`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "source-system-auth-token",
+		Target:  &c.sourceToken,
+		Aliases: []string{"st", "src-token"},
+		Usage:   `Token to authenticate with source system to read membership information`,
+	})
+
+	f.StringVar(&cli.StringVar{
+		Name:    "des-system-auth-token",
+		Target:  &c.destinationToken,
+		Aliases: []string{"dt", "dst-token"},
+		Example: "user-mapping-config.textproto",
+		Usage:   `Token to authenticate with destination system to write membership information`,
+	})
+
+	return set
+}
+
+func (c *SyncCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+	args = f.Args()
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %q", args)
+	}
+
+	if _, ok := slices.BinarySearch(allowedSourceSystem, strings.ToUpper(c.source)); !ok {
```

Comment:
`BinarySearch` requires `allowedSourceSystem` to be in sorted order, and when we add more source systems in the future I'm not sure that other devs will remember to keep that list sorted. 
Also, I doubt the number of source or destination systems will ever be greater than ~10, so I think using linear search (`slices.Contains`) would be the better choice for simplicity.

Reply:
+1 for slices.Contains

Reply:
Fixed. Thank you!